extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -119,16 +120,29 @@ public void handle(AsyncResult<OAuth2Auth> event) {
 
                 auth = cf.join();
                 break;
-            } catch (OIDCException ex) {
-                if (i + 1 < connectionRetryCount) {
-                    try {
-                        Thread.sleep(2000);
-                    } catch (InterruptedException iex) {
-                        // continue connecting
+            } catch (Throwable throwable) {
+                while (throwable instanceof CompletionException && throwable.getCause() != null) {
+                    throwable = throwable.getCause();
+                }
+                if (throwable instanceof OIDCException) {
+                    if (i + 1 < connectionRetryCount) {
+                        CompletableFuture.runAsync(
+                                new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        try {
+                                            Thread.sleep(2000);
+                                        } catch (InterruptedException iex) {
+                                            // continue connecting
+                                        }
+                                    }
+                                })
+                                .join();
```

Comment:
Also this blocks the thread anyway, so now you are blocking 2 threads instead of just one.

Reply:
Well it can be concluded I did not draw the right conclusion based on the `io thread` proposal :-)

Reply:
I misread this, I thought it was happening at runtime. It's not ideal that it connects to each tenant sequentially but it is probably fine for now, sorry for the noise.

Reply:
@stuartwdouglas Hi Stuart, no problems at all, it has been awhile since I had to think so hard, trying to figure out how to link all those Functions and Consumers :-).

Good point about the runtime, in fact we have 2 cases here, one is when the static TenantContextConfigs are created, those in the application.properties and a new case when the multi-tenancy resolvers may need it, during the client request, and I was thinking may be we should not even support the `connection-delay` property in this latter case, if someone sets to even 10 sec it won't be good. This property has been added to help in the cases when at the initial start up, the OIDC server is a few secs behind, to help with the tests mainly...I'll create a small issue afterwards, so that we can decide if to ignore this property in those cases or if not then at least make the more optimal use of the current io thread as per proposal, etc